content/Methods/Methods.q.md

```diff
@@ -0,0 +1,1329 @@
+
+ - syntax error, unexpected `'changeName'` (T_STRING), expecting variable (T_VARIABLE) on line number 6
+
+:::
+
+
+:::
+
+/// type=REPL
+
+```php
+<?php
+    class Person 
+    {
+        public $name = "Diana";
+
+        public function changeName($newName)
+        {
+            ->name = $newName;
+        }
+
+        public function display
+        {
+            echo "My name is " . $this->name . ".";
+        }
+    }
+    $personObject = new Person();
+    $personObject->changeName("Charles");
+    $personObjectdisplay();
+?>
+```
+/// type=SS, answer=[3]
+
+In the property definition `public $name = "Diana";` on line 4, remove the dollar sign `$` from `$name`. Execute the program. What is the error message?
+
+ - syntax error, unexpected `'"'`, expecting `','` or `';'` on line number 4
+
+ - syntax error, unexpected `'public'` (T_PUBLIC), expecting `'{'` on line number 4
+
+ - syntax error, unexpected `'name'` (T_STRING), expecting variable (T_VARIABLE) on line number 4
+
+ - syntax error, unexpected `'$name'` (T_VARIABLE), expecting function (T_FUNCTION) on line number 4
+
+ -  syntax error, unexpected `'"Diana"'` (T_CONSTANT_ENCAPSED_STRING), expecting `','` or `';'` on line number 4
+
+:::
+
+
+/// type=CR, answer=[tests/Methods/CorrectMultipleErrorsTest.php]
+
+Correct the code so that it outputs the string `My name is Charles.`.
+
+```php
+<?php
+    class Person 
+    {
+        public name = "Diana";
+
+        public function changeName($newName)
+        {
+            ->name = $newName;
+        }
+
+        public function display
+        {
+            echo "My name is " . $this->name . ".";
+        }
+    }
+    $personObject = new Person();
+    $personObject->changeName("Charles");
+    $personObjectdisplay();
+?>
+```
+
++++
+
+
++++
+
+### Part 4: Practice
+
+/// type=CR, answer=[tests/Methods/ManipulateObjectMethodsTest.php]
+
+Write a program that adds and manipulates the methods of a certain class. First, use a `class` keyword to declare a class named `Animal`. Within the curly braces `{}`, add a property definition of a class property `$type` with the value `Dog`. Next, add a method definition for `changeType()` method with a parameter named `$newType`. Inside the `changeType()` method body, add a statement that assigns the value of `$newType` to the `$type` property of the `Animal` class. Add another method definition for a method named `display()`. Inside the `display()` method body, add an `echo` statement to display the string `"The animal type is " . $this->type . "."`. After the class declaration, add a statement that creates the `$pet` object an instance of the `Animal` class. Add a statement that calls the `changeType()` method of the `$pet` object passing the argument `Cat`. Then, add another statement that calls the `display()` method of the `$pet` object. Run the program to view the output.
+
+```php
+<?php
```

Comment:
It might be helpful to demonstrate chained method calls within an object to clarify how $this works. I'm not sure if that would be considered practice or something done in part 1-3? Or both?

Reply:
This is noted, T. I will try to revisit the entire lesson content and check where I can include this information. Thank you.